src/cpp/core/system/PosixSystem.cpp

```diff
@@ -2426,11 +2429,63 @@ Error runProcess(const std::string& path,
    return error;
 }
 
-Error getChildProcesses(std::vector<ProcessInfo> *pOutProcesses, bool populateUsername)
+Error getChildProcesses(
+      std::vector<ProcessInfo>* pOutProcesses,
+      bool populateUsername)
 {
    return getChildProcesses(::getpid(), pOutProcesses, populateUsername);
 }
 
+#ifdef __APPLE__
+
+Error getChildProcesses(
+      pid_t pid,
+      std::vector<ProcessInfo>* pOutProcesses,
+      bool populateUsername)
+{
+   if (!pOutProcesses)
+      return systemError(EINVAL, ERROR_LOCATION);
+
+   // get child processes
+   const std::size_t N = 1024;
+   std::vector<pid_t> pids(N);
+   int count = proc_listchildpids(pid, pids.data(), N * sizeof(pid_t));
+   if (count == -1)
+      return systemError(errno, ERROR_LOCATION);
+   
+   // build process info
+   std::vector<ProcessInfo> processes;
+   for (auto i = 0; i < count; i++)
+   {
+      // Get process info for each child
+      struct proc_bsdshortinfo procInfo;
+      int result = proc_pidinfo(pids[i], PROC_PIDT_SHORTBSDINFO, 0, &procInfo, sizeof(procInfo));
+      if (result == -1)
+         continue;
+      
+      ProcessInfo info;
+      info.pid = procInfo.pbsi_pid;
+      info.ppid = procInfo.pbsi_ppid;
+      info.pgrp = procInfo.pbsi_gid;
+      info.exe = procInfo.pbsi_comm;
+      info.uid_ = procInfo.pbsi_uid;
+      info.uidSet_ = true;
+      
+      if (populateUsername)
+      {
+         struct passwd* pwd = getpwuid(procInfo.pbsi_uid);
```

Comment:
Could we use: core::system::getUserFromUserId(..) here?   It uses a cache to prevent lots of calls to getpwuid, each of which will turn into possibly an ldap or now user service api call.